File: app/src/main/java/me/anwarshahriar/rxjavademo/view/GistView.java
package me.anwarshahriar.rxjavademo.view;

import java.util.List;
import me.anwarshahriar.rxjavademo.model.Gist;

public interface GistView {
  void showEmptyUsernameError(String message);
  void showNoGistsStatus();
  void hideNoGistsStatus();
  void showLoading();
  void hideLoading();
  String getUsername();
  void showGists(List<Gist> gists);
}

File: app/src/main/java/me/anwarshahriar/rxjavademo/view/GistView.kt
package me.anwarshahriar.rxjavademo.view

import me.anwarshahriar.rxjavademo.model.Gist

interface GistView {
  fun showEmptyUsernameError(message: String)
  fun showNoGistsStatus()
  fun hideNoGistsStatus()
  fun showLoading()
  fun hideLoading()
  fun getUsername(): String?
  fun showGists(gists: List<Gist>)
}
